docs/core/diagnostics/metrics-instrumentation.md

```diff
@@ -471,3 +524,86 @@ Press p to pause, r to resume, q to quit.
 
 > [!NOTE]
 > OpenTelemetry refers to tags as 'attributes'. These are two different names for the same functionality.
+
+## Test custom metrics
+
+Its possible to test any custom metrics you add using <xref:Microsoft.Extensions.Telemetry.Testing.Metering.MetricCollector%601>. This type makes it easy to record the measurements
+from specific instruments and assert the values were correct.
+
+### Test with dependency injection
+
+The following code shows an example test case for code components that use dependency injection and IMeterFactory.
+
+```csharp
+public class MetricTests
+{
+    [Fact]
+    public void SaleIncrementsHatsSoldCounter()
+    {
+        // Arrange
+        var services = CreateServiceProvider();
+        var metrics = services.GetRequiredService<HatCoMetrics>();
+        var meterFactory = services.GetRequiredService<IMeterFactory>();
+        var collector = new MetricCollector<int>(meterFactory, "HatCo.Store", "hatco.store.hats_sold");
+
+        // Act
+        metrics.HatsSold(15);
+
+        // Assert
+        var measurements = collector.GetMeasurementSnapshot();
+        Assert.Equal(1, measurements.Count);
+        Assert.Equal(15, measurements[0].Value);
+    }
+
+    // Setup a new service provider. This example creates the collection explicitly but you might leverage
+    // a host or some other application setup code to do this as well.
+    private static IServiceProvider CreateServiceProvider()
+    {
+        var serviceCollection = new ServiceCollection();
+        serviceCollection.AddMetrics();
+        serviceCollection.AddSingleton<HatCoMetrics>();
+        return serviceCollection.BuildServiceProvider();
+    }
+}
+```
+
+Each MetricCollector object records all measurements for one Instrument. If you need to verify measurements from multiple instruments, create one MetricCollector for each one.
+
+### Test without dependency injection
+
+It is also possible to test code that uses a shared global Meter object in a static field, but make sure such tests are configured not to run in parallel. Because the
+Meter object is being shared, MetricCollector in one test will observe the measurements created from any other tests running in parallel.
+
+```csharp
+class HatCoMetricsWithGlobalMeter
+{
+    static Meter s_meter = new Meter("HatCo.Store");
+    static Counter<int> s_hatsSold = s_meter.CreateCounter<int>("hatco.store.hats_sold");
+
+    public void HatsSold(int quantity)
+    {
+        s_hatsSold.Add(quantity);
+    }
+}
+
+public class MetricTests
+{
+    [Fact]
+    public void SaleIncrementsHatsSoldCounter()
+    {
+        // Arrange
+        var metrics = new HatCoMetricsWithGlobalMeter();
+        // Be careful specifying scope=null. This binds the collector to a global Meter and tests
+        // that use global state should not be configured to run in parallel.
+        var collector = new MetricCollector<int>(null, "HatCo.Store", "hatco.store.hats_sold");
```

Comment:
Should `MetricCollector` have a ctor that doesn't have a scope argument?

Reply:
It does have ctors like (Meter, string) and (Instrument), but it was intentional not to offer a ctor that takes a meter name string binds in global scope by default.